cmd/otel-agent/config/agent_config.go

```diff
@@ -72,8 +72,29 @@ func NewConfigComponent(ctx context.Context, uris []string) (config.Component, e
 	pkgconfig.Set("logs_enabled", true, pkgconfigmodel.SourceLocalConfigProcess)
 	pkgconfig.Set("logs_config.use_compression", true, pkgconfigmodel.SourceLocalConfigProcess)
 	pkgconfig.Set("log_level", sc.Telemetry.Logs.Level, pkgconfigmodel.SourceLocalConfigProcess)
+
+	// APM & OTel trace configs
 	pkgconfig.Set("apm_config.enabled", true, pkgconfigmodel.SourceLocalConfigProcess)
 	pkgconfig.Set("apm_config.apm_non_local_traffic", true, pkgconfigmodel.SourceLocalConfigProcess)
+	pkgconfig.Set("otlp_config.traces.span_name_as_resource_name", ddc.Traces.SpanNameAsResourceName, pkgconfigmodel.SourceLocalConfigProcess)
```

Comment:
set this to true by default

Reply:
No we cannot do it until the OSS datadog connector sets this to true. Right now this is false by default in datadog connector. If here it is set to true, the span names in spans vs. in APM stats will be different and that leads to mismatch or missing spans on the UI.

Reply:
Chatted offline, I flipped compute_top_level_by_span_kind and span_name_as_resource_name to true by default. 

We need to make sure the datadog connector has the same default in OTel agent (via config converter?).  cc @mackjmr